File: distribution/python/dqops/client/models/incident_daily_issues_count.py
from typing import Any, Dict, List, Type, TypeVar, Union

from attrs import define as _attrs_define
from attrs import field as _attrs_field

from ..types import UNSET, Unset

T = TypeVar("T", bound="IncidentDailyIssuesCount")


@_attrs_define
class IncidentDailyIssuesCount:
    """
    Attributes:
        warnings (Union[Unset, int]): The number of failed data quality checks that generated a warning severity data
            quality issue.
        errors (Union[Unset, int]): The number of failed data quality checks that generated an error severity data
            quality issue.
        fatals (Union[Unset, int]): The number of failed data quality checks that generated a fatal severity data
            quality issue.
        total_count (Union[Unset, int]): The total count of failed data quality checks on this day.
    """

    warnings: Union[Unset, int] = UNSET
    errors: Union[Unset, int] = UNSET
    fatals: Union[Unset, int] = UNSET
    total_count: Union[Unset, int] = UNSET
    additional_properties: Dict[str, Any] = _attrs_field(init=False, factory=dict)

    def to_dict(self) -> Dict[str, Any]:
        warnings = self.warnings
        errors = self.errors
        fatals = self.fatals
        total_count = self.total_count

        field_dict: Dict[str, Any] = {}
        field_dict.update(self.additional_properties)
        field_dict.update({})
        if warnings is not UNSET:
            field_dict["warnings"] = warnings
        if errors is not UNSET:
            field_dict["errors"] = errors
        if fatals is not UNSET:
            field_dict["fatals"] = fatals
        if total_count is not UNSET:
            field_dict["totalCount"] = total_count

        return field_dict

    @classmethod
    def from_dict(cls: Type[T], src_dict: Dict[str, Any]) -> T:
        d = src_dict.copy()
        warnings = d.pop("warnings", UNSET)

        errors = d.pop("errors", UNSET)

        fatals = d.pop("fatals", UNSET)

        total_count = d.pop("totalCount", UNSET)

        incident_daily_issues_count = cls(
            warnings=warnings,
            errors=errors,
            fatals=fatals,
            total_count=total_count,
        )

        incident_daily_issues_count.additional_properties = d
        return incident_daily_issues_count

    @property
    def additional_keys(self) -> List[str]:
        return list(self.additional_properties.keys())

    def __getitem__(self, key: str) -> Any:
        return self.additional_properties[key]

    def __setitem__(self, key: str, value: Any) -> None:
        self.additional_properties[key] = value

    def __delitem__(self, key: str) -> None:
        del self.additional_properties[key]

    def __contains__(self, key: str) -> bool:
        return key in self.additional_properties

File: dqops/src/main/java/com/dqops/data/incidents/models/IncidentDailyIssuesCount.java
/*
 * Copyright © 2021 DQOps (support@dqops.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.dqops.data.incidents.models;

import com.dqops.utils.docs.generators.SampleValueFactory;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.Data;

import java.time.LocalDateTime;
import java.time.ZoneOffset;

/**
 * A model that stores a daily number of incidents.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonNaming(PropertyNamingStrategies.LowerCamelCaseStrategy.class)
@Data
public class IncidentDailyIssuesCount {
    /**
     * The number of failed data quality checks that generated a warning severity data quality issue.
     */
    @JsonPropertyDescription("The number of failed data quality checks that generated a warning severity data quality issue.")
    private int warnings = 0;

    /**
     * The number of failed data quality checks that generated an error severity data quality issue.
     */
    @JsonPropertyDescription("The number of failed data quality checks that generated an error severity data quality issue.")
    private int errors = 0;

    /**
     * The number of failed data quality checks that generated a fatal severity data quality issue.
     */
    @JsonPropertyDescription("The number of failed data quality checks that generated a fatal severity data quality issue.")
    private int fatals = 0;

    /**
     * The total count of failed data quality checks on this day.
     */
    @JsonPropertyDescription("The total count of failed data quality checks on this day.")
    private int totalCount = 0;

    /**
     * Increment the count of failed data quality checks for a given severity level.
     * @param severityLevel Severity level of a failed data quality check.
     */
    public void incrementForIssueSeverity(int severityLevel) {
        this.totalCount++;

        if (severityLevel == 1) {
            this.warnings++;
        }
        else if (severityLevel == 2) {
            this.errors++;
        }
        else if (severityLevel == 3) {
            this.fatals++;
        }
    }

    /**
     * Sample factory for an incident model.
     */
    public static class IncidentDailyIssuesCountSampleFactory implements SampleValueFactory<IncidentDailyIssuesCount> {
        @Override
        public IncidentDailyIssuesCount createSample() {
            return new IncidentDailyIssuesCount() {{
                setTotalCount(10);
                setWarnings(2);
                setErrors(7);
                setFatals(1);
            }};
        }
    }
}
